src/iris/vendors/iris_slack.py

```diff
@@ -95,17 +95,27 @@ def send_message(self, message):
         payload = self.get_message_payload(message)
         try:
             response = requests.post(self.config['base_url'],
-                                     data=payload,
+                                     data=ujson.dumps(payload),
                                      proxies=self.proxy,
                                      timeout=self.timeout)
             if response.status_code == 200:
-                data = response.json()
-                if data['ok']:
-                    return time.time() - start
-                # If message is invalid:
-                #   {u'ok': False, u'error': u'invalid_arg_name'}
-                logger.error('Received an error from slack api: %s',
-                             data['error'])
+                try:
+                    data = response.json()
+                    if data['ok']:
+                        return time.time() - start
+                    # If message is invalid:
+                    #   {u'ok': False, u'error': u'invalid_arg_name'}
+                    logger.error('Received an error from slack api: %s',
+                                 data['error'])
+                except Exception:
```

Comment:
It looks like Mattermost's API is largely compatible with Slack's, but I don't think this is a good way to handle the incompatibilities. Can you make a new vendor instead? You can duplicate some of this code, but adapt it to better suit Mattermost (get rid of the buttons and handle the different response)